src/util/strencodings.h

```diff
@@ -305,4 +305,17 @@ std::string ToUpper(const std::string& str);
  */
 std::string Capitalize(std::string str);
 
+/**
+ * Parse a string with suffix unit [k|K] [m|M] [g|G] [t|T] .
+ * Must be a whole integer, fractions not allowed (0.5t).
+ * Lowercase units are 1000 base. Uppercase units are 1024 base.
+ * Examples: 2m,27M,19g,41T
+ *
+ * @param[in] str                  the string to convert into bytes
+ * @param[in] default_multiplier   if no unit is found in str, use this multiplier. default is MiB
+ * @returns                        optional uint64_t bytes from str or nullopt
+ *                                 if ParseUInt64 is false, str is empty, trailing whitespace or overflow
+ */
+std::optional<uint64_t> ParseByteUnits(const std::string& str, const uint64_t default_multiplier = 1024UL*1024UL);
```

Comment:
nit: drop unnecessary `const` from the integer argument
style: missing spaces around `*`:

```suggestion
std::optional<uint64_t> ParseByteUnits(const std::string& str, uint64_t default_multiplier = 1024UL * 1024UL);
```

Reply:
The `const` is still there. I think there was discussion somewhere about it, but I can't find it now...